core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -125,6 +125,13 @@ public void setTermsLookupFetchService(@Nullable  TermsLookupFetchService termsL
         this.termsLookupFetchService = termsLookupFetchService;
     }
 
+    /**
+     * For testing purposes only - ensures that e.g. for GeoBoundingBoxQueries we can inject a mocked service.
+     * */
+    void setFieldDataService(IndexFieldDataService fieldDataService) {
+        this.fieldDataService = fieldDataService;
+    }
```

Comment:
not loving this setter for testing only. Do you mind moving the mock service to BaseQueryTestCase, and binding it through guice? just need to add a new line `bind(IndexFieldDataService.class).to(MockIndexFieldDataService.class);` to the base class, when we initialize the injector:

```
new AbstractModule() {
                    @Override
                    protected void configure() {
                        Multibinder.newSetBinder(binder(), ScoreFunctionParser.class);
                        bind(ClusterService.class).toProvider(Providers.of(clusterService));
                        bind(CircuitBreakerService.class).to(NoneCircuitBreakerService.class);
                        bind(IndexFieldDataService.class).to(MockIndexFieldDataService.class);
                        bind(NamedWriteableRegistry.class).asEagerSingleton();
                    }
                }
```